### PR TITLE
Add WasmEdge Style Guide

### PR DESCRIPTION
**Pull request by mhmohona**
## Explanation

This PR introduces a new file, `STYLE_GUIDE.md`, to the WasmEdge repository. This file contains two style guides: one for documentation and one for coding. 

## Related issue
https://github.com/WasmEdge/docs/issues/85

## What type of PR is this
/kind documentation

### CODE REVIEW

**Comment by alabulei1** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: docs/develop/getting-started/faq.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+WasmEdge follows the WebAssembly specification, which currently does not support shared memory between different modules. Each module has its own linear memory space.
+
+## 2. Can WasmEdge support model training?
+
+Currently, WasmEdge supports model inference. It uses the WASI-NN API to make predictions using pre-trained models. However, model training is not yet supported.
+
+## 3. What is the internal flow of WasmEdge?
+
+The WasmEdge runtime follows a general flow: parsing the Wasm file, validating the parsed Wasm file, compiling the validated Wasm file into native code, and then executing the compiled code. For more detailed information, please refer to the WasmEdge runtime architecture documentation.
```
Comment (alabulei1) [inline docs/develop/getting-started/faq.md:19]: Link to our docs.

File: docs/develop/getting-started/faq.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+This FAQ page is designed to address the most common technical inquiries about WasmEdge. If your question is not directly answered here, please refer to the comprehensive WasmEdge documentation or engage with the active WasmEdge community.
+
+## 1. How does WasmEdge handle memory sharing between modules?
+
+WasmEdge follows the WebAssembly specification, which currently does not support shared memory between different modules. Each module has its own linear memory space.
+
+## 2. Can WasmEdge support model training?
+
+Currently, WasmEdge supports model inference. It uses the WASI-NN API to make predictions using pre-trained models. However, model training is not yet supported.
```
Comment (alabulei1) [inline docs/develop/getting-started/faq.md:15]: Link an article for AI/LLM inference.

File: docs/develop/getting-started/faq.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+## 3. What is the internal flow of WasmEdge?
+
+The WasmEdge runtime follows a general flow: parsing the Wasm file, validating the parsed Wasm file, compiling the validated Wasm file into native code, and then executing the compiled code. For more detailed information, please refer to the WasmEdge runtime architecture documentation.
+
+## 4. Why is my plugin crashing?
+
+If your plugin crashes, it might be due to several reasons. It could be related to incorrect use of the WasmEdge API, or the plugin may be incompatible with the WasmEdge version you're using. It's recommended to debug the plugin using a debugger tool to get more detailed error information.
```
Comment (alabulei1) [inline docs/develop/getting-started/faq.md:23]: how to debug? Add a link here.

File: docs/develop/getting-started/faq.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+## 5. How to create a VM to call `infer()` in a Wasm library?
+
+You can use the WASI-NN API to call the `infer()` function in a Wasm library. First, you need to prepare the model, inputs, and outputs. Then, you can call the `infer()` function with these parameters.
+
+## 6. Can WasmEdge support Tensorflow as its inference backend using WASI-NN?
+
+Yes, WasmEdge can use Tensorflow as its inference backend through the WASI-NN API.
```
Comment (alabulei1) [inline docs/develop/getting-started/faq.md:31]: Add a link for tensorflow inference

File: docs/develop/getting-started/faq.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+## 7. How to read a host file in WasmEdge runtime?
+
+WasmEdge provides the WASI (WebAssembly System Interface) API for interacting with the host system, including file operations. You can use the WASI API to open and read files from the host system.
```
Comment (alabulei1) [inline docs/develop/getting-started/faq.md:35]: Link a doc for this function.

File: docs/develop/getting-started/faq.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+## 8. Does WasmEdge only support one backend at the same time?
```
Comment (alabulei1) [inline docs/develop/getting-started/faq.md:37]: What do you mean by backend?

**Comment by alabulei1** [comment]
Hi  @mhmohona 

Please check out my comments. Thanks. BTW, how did you find these issues?

File: docs/develop/getting-started/faq.md (hunk 2 above)
Comment (mhmohona) [reply docs/develop/getting-started/faq.md:15]: Could you please provide the link of that article? I am confused which one should i add.

Comment (alabulei1) [reply docs/develop/getting-started/faq.md]: https://www.secondstate.io/articles/fast-llm-inference/

**Comment by mhmohona** [comment]
@alabulei1 so here are the source, I had these questions while writing the plugin doc. ALso i went through the [discussion](https://github.com/WasmEdge/WasmEdge/discussions) to find the general questions about plugin.

**Pull request closed without merging**